crates/uv/tests/it/sync.rs

```diff
@@ -11037,7 +11055,7 @@ fn transitive_group_conflicts_shallow() -> Result<()> {
 
     ----- stderr -----
     Resolved 5 packages in [TIME]
-    error: Groups `dev` and `magic` are incompatible with the transitively inferred conflicts: {`example:dev`, `example:magic`}
+    error: Groups `dev` and `magic` are incompatible with the declared conflicts: {`example:dev`, `example:magic`}
```

Comment:
Unfortunately this regresses the error message, I'll need to investigate that further.

Reply:
Presumably the problem is that we're installing from the lockfile instead of a fresh resolution, which drops this information.

I think there are a few options here

1. Read the pyproject.toml to determine if the conflict is transitive
2. Change the encoding in the lockfile to avoid flattening nested groups
3. Add a marker to the lockfile to indicate a transitive conflict is present
4. Drop this special-case entirely

I sort of argue for (4), while we might want (2) or (3) in the future, they seem like a fair bit of work and the improvement in the error message is quite minor.